File: Placement Sheet/LinkedList/PairWiseSwap.java
public class PairWiseSwap {
    public static class Node {
        int data;
        Node next;

        // Constructor...
        Node(int data) {
            this.data = data;
            this.next = next;
        }
    }

    private Node head = null;
    private Node tail = null;

    // Insert...
    public void insert(int data) {
        Node newNode = new Node(data);
        if (head == null) {
            head = newNode;
            tail = newNode;
            return;
        }
        Node currNode = head;
        while (currNode.next != null) {
            currNode = currNode.next;
        }
        currNode.next = newNode;
        tail.next = newNode;
        tail = newNode;
    }

    // Display List...
    public void display() {
        if (head == null) {
            System.out.println("List is empty...");
            return;
        }
        Node currNode = head;
        while (currNode != null) {
            System.out.print(currNode.data + " --> ");
            currNode = currNode.next;
        }
        System.out.println("Null");
    }

    // Swapping...
    public void pairSwap() {
        if (head == null) {
            System.out.println("List is empty...");
            return;
        }
        Node dummyNode = new Node(0);
        dummyNode.next = head;
        Node prevNode = dummyNode;

        while (prevNode.next != null && prevNode.next.next != null) {
            Node firsNode = prevNode.next;
            Node secondNode = firsNode.next;

            firsNode.next = secondNode.next;
            secondNode.next = firsNode;
            prevNode.next = secondNode;
            prevNode = firsNode;
        }
        head = dummyNode.next;
    }

    public static void main(String[] args) {
        PairWiseSwap list = new PairWiseSwap();
        list.insert(1);
        list.insert(2);
        list.insert(3);
        list.insert(4);
        list.insert(5);
        list.insert(6);
        System.out.print("Before swapping: ");
        list.display();
        System.out.println();
        System.out.println("After swapping: ");
        list.pairSwap();
        list.display();
    }
}

File: Placement Sheet/LinkedList/BS/PairWiseSwap.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var ListNode = /** @class */ (function () {
    function ListNode(data) {
        this.data = data;
        this.next = null;
    }
    ListNode.prototype.getData = function () {
        return this.data;
    };
    ListNode.prototype.getNext = function () {
        return this.next;
    };
    ListNode.prototype.setNext = function (node) {
        this.next = node;
    };
    return ListNode;
}());
var LinkedList = /** @class */ (function () {
    function LinkedList() {
        this.head = null;
    }
    //Insert...
    LinkedList.prototype.append = function (data) {
        var newNode = new ListNode(data);
        if (this.head == null) {
            this.head = newNode;
            return;
        }
        var currNode = this.head;
        while (currNode.getNext() !== null) {
            currNode = currNode === null || currNode === void 0 ? void 0 : currNode.getNext();
        }
        currNode.setNext(newNode);
    };
    //Display the List...
    LinkedList.prototype.display = function () {
        if (this.head == null) {
            console.log("List is empty...");
            return;
        }
        var currNode = this.head;
        var result = "";
        while (currNode !== null) {
            result += currNode.getData() + " --> ";
            currNode = currNode.getNext();
        }
        result += "Null";
        console.log(result);
    };
    //Swapping...
    LinkedList.prototype.pairSwap = function () {
        var _a;
        var dummyNode = new ListNode(0);
        dummyNode.setNext(this.head);
        var prevNode = dummyNode;
        while (prevNode.getNext() && ((_a = prevNode.getNext()) === null || _a === void 0 ? void 0 : _a.getNext())) {
            var first = prevNode.getNext();
            var second = first === null || first === void 0 ? void 0 : first.getNext();
            first === null || first === void 0 ? void 0 : first.setNext(second === null || second === void 0 ? void 0 : second.getNext());
            second === null || second === void 0 ? void 0 : second.setNext(first);
            prevNode.setNext(second);
            prevNode = first;
        }
        this.head = dummyNode.getNext();
    };
    return LinkedList;
}());
var list = new LinkedList();
list.append(1);
list.append(2);
list.append(3);
list.append(4);
console.log("Before Swapping...");
list.display();
console.log("After Swapping...");
list.pairSwap();
list.display();
